src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -296,5 +284,140 @@ internal SafeSslHandle(IntPtr validSslPointer, bool ownsHandle) : base(IntPtr.Ze
         {
             handle = validSslPointer;
         }
+
+        internal class ReadBioBuffer : IDisposable
+        {
+            private SafeBioHandle _bioHandle;
+            private GCHandle _handle;
+            private int _bytesAvailable;
+            private byte[] _byteArray;
+            private int _offset;
+
+            internal ReadBioBuffer(SafeBioHandle bioHandle)
+            {
+                _bioHandle = bioHandle;
+                _handle = GCHandle.Alloc(this, GCHandleType.Normal);
+                Interop.CustomBio.BioSetGCHandle(_bioHandle, _handle);
+                Interop.Crypto.BioSetShoudRetryReadFlag(bioHandle);
+            }
+
+            public void SetBio(byte[] buffer, int offset, int length)
```

Comment:
`SetBio` doesn't feel like it matches what's happening.  You're not setting an underlying BIO, but rather just setting the memory to be read.  `SetData` seems more apt.